mbedtls/src/pk/mod.rs

```diff
@@ -178,9 +184,124 @@ impl Pk {
+            Type::Eckey | Type::EckeyDh | Type::Ecdsa => {}
+            _ => return Err(::Error::PkTypeMismatch),
+        }
+
+        let q = &unsafe { (*(self.inner.pk_ctx as *const ecp_keypair)).Q };
+        EcPoint::copy(q)
+    }
+
+    pub fn ec_private(&self) -> ::Result<Mpi> {
+        match self.pk_type() {
+            Type::Eckey | Type::EckeyDh | Type::Ecdsa => {}
+            _ => return Err(::Error::PkTypeMismatch),
+        }
+
+        let d = &unsafe { (*(self.inner.pk_ctx as *const ecp_keypair)).d };
+        Mpi::copy(d)
+    }
+
+    pub fn rsa_public_modulus(&self) -> ::Result<Mpi> {
+        match self.pk_type() {
+            Type::Rsa => {}
+            _ => return Err(::Error::PkTypeMismatch),
+        }
+
+        let mut n = Mpi::new(0)?;
+
+        unsafe {
+            rsa_export(
+                self.inner.pk_ctx as *const rsa_context,
+                n.handle_mut(),
+                ptr::null_mut(),
+                ptr::null_mut(),
+                ptr::null_mut(),
+                ptr::null_mut(),
+            )
+            .into_result()?;
+        }
+
+        Ok(n)
+    }
+
+    pub fn rsa_private_prime1(&self) -> ::Result<Mpi> {
+        match self.pk_type() {
+            Type::Rsa => {}
+            _ => return Err(::Error::PkTypeMismatch),
+        }
+
+        let mut p = Mpi::new(0)?;
+
+        unsafe {
+            rsa_export(
+                self.inner.pk_ctx as *const rsa_context,
+                ptr::null_mut(),
+                p.handle_mut(),
+                ptr::null_mut(),
+                ptr::null_mut(),
+                ptr::null_mut(),
+            )
+            .into_result()?;
+        }
+
+        Ok(p)
+    }
+
+    pub fn rsa_private_prime2(&self) -> ::Result<Mpi> {
+        match self.pk_type() {
+            Type::Rsa => {}
+            _ => return Err(::Error::PkTypeMismatch),
+        }
+
+        let mut q = Mpi::new(0)?;
+
+        unsafe {
+            rsa_export(
+                self.inner.pk_ctx as *const rsa_context,
+                ptr::null_mut(),
+                ptr::null_mut(),
+                q.handle_mut(),
+                ptr::null_mut(),
+                ptr::null_mut(),
+            )
+            .into_result()?;
+        }
+
+        Ok(q)
+    }
+
+    pub fn rsa_public_exponent(&self) -> ::Result<u32> {
+        match self.pk_type() {
+            Type::Rsa => {}
+            _ => return Err(::Error::PkTypeMismatch),
+        }
+
+        let mut e: [u8; 4] = [0, 0, 0, 0];
+        unsafe {
+            rsa_export_raw(
+                self.inner.pk_ctx as *const rsa_context,
+                ptr::null_mut(),
+                0,
+                ptr::null_mut(),
+                0,
+                ptr::null_mut(),
+                0,
+                ptr::null_mut(),
+                0,
+                e.as_mut_ptr(),
+                e.len(),
+            )
+            .into_result()?;
+        }
+        Ok(BigEndian::read_u32(&e))
```

Comment:
[`u32::from_be_bytes`](https://doc.rust-lang.org/std/primitive.u32.html#method.from_be_bytes) is stable now, using that avoids byteorder dependency.

Reply:
It is stable, but is not in the old nightly used for the core_io build.

Reply:
Leaving byteorder for now, we can address this when core_io build is updated to a recent nightly. I would take this on, but I am not sure what is needed.

Reply:
> It is stable, but is not in the old nightly used for the core_io build.

Oh yeah :( never mind then